rocksdb_admin/admin_handler.cpp

```diff
@@ -81,6 +89,51 @@ namespace {
 const int kMB = 1024 * 1024;
 const int kS3UtilRecheckSec = 5;
 
+const int64_t kMillisPerSec = 1000;
+const char kKafkaConsumerType[] = "rocksplicator_message_consumer";
+const char kKafkaWatcherName[] = "RocksplicatorMessageConsumer";
+const uint32_t kKafkaConsumerPoolSize = 1;
+
+// Global map to keep track of serverset file watchers
+std::unordered_map<std::string, std::shared_ptr<
+    ::kafka::KafkaBrokerFileWatcher>> serverset_path_to_file_watcher_map;
+
+// Global condition variables for kafka synchronization.
+// TODO (rajath): add these to a class so it's reusable
+struct SyncVariables
+{
+  SyncVariables(): flag(false) {} // init flag to false
+  bool flag;
+  std::condition_variable cond_var;
+  std::mutex mutex;
+};
+std::unordered_map<std::string, std::shared_ptr<SyncVariables>> sync_var_map;
+std::mutex sync_var_map_mutex;
+std::mutex serverset_path_to_file_watcher_map_mutex;
+
+int64_t GetMessageTimestampSecs(const RdKafka::Message& message) {
+  const auto ts = message.timestamp();
+  if (ts.type == RdKafka::MessageTimestamp::MSG_TIMESTAMP_CREATE_TIME) {
+    return ts.timestamp / kMillisPerSec;
+  }
+
+  // We only expect the timestamp to be create time.
+  return -1;
+}
+
+std::string ToUTC(const int64_t time_secs) {
+  time_t raw_time(time_secs);
+  char buf[24];
```

Comment:
should be, since the number of characters is fixed. I've changed it to 48 to be safe. I'll most likely be getting rid of this function before productionizing it.